lib/jekyll-redirect-from/redirector.rb

```diff
@@ -71,7 +71,14 @@ def redirect_url(site, item)
     end
 
     def redirect_prefix(site)
-      config_github_url(site) || config_url(site)
+      config_url_and_baseurl(site) || config_github_url(site) || config_baseurl(site) || ""
+    end
+
+    def config_url_and_baseurl(site)
+      return unless config_url(site)
+
+      prefix = File.join(config_url(site).to_s, config_baseurl(site).to_s)
```

Comment:
Makes sense. Can change all the `File.joins` in a subsequent pass if we want. I knew I remembered a discussion about this...